From 20bf13de482b02a4a467f44070f7ff184c340dd2 Mon Sep 17 00:00:00 2001
From: Jayaprakash Mutyala <mutyalax.jayaprakash@intel.com>
Date: Tue, 1 Jun 2021 12:22:19 +0000
Subject: [PATCH] rakp12: Add username to SessionInfo interface

Add username to SessionInfo interface to get username info on individual
IPMI session Id's on Redfish

Tested:
1. Activate SOL session
Command: ipmitool -I lanplus -U root -P 0penBmc -H <BMC_IP> -C 17
         sol activate
Response:                         // Success
2. Verified on SessionInfo D-bus interface.
busctl introspect xyz.openbmc_project.Ipmi.Channel.eth0
                  /xyz/openbmc_project/ipmi/session/eth0/<session_id>
NAME               TYPE      SIGNATURE RESULT/VALUE FLAGS
......
xyz.openbmc_project.Ipmi.SessionInfo interface -         -            -
.ChannelNum          property  y      3            emits-change writable
.CurrentPrivilege    property  y      4            emits-change writable
.RemoteIPAddr        property  u      22253066     emits-change writable
.RemoteMACAddress    property  ay     0            emits-change writable
.RemotePort          property  q      41096        emits-change writable
.SessionHandle       property  y      129          emits-change writable
.State               property  y      2            emits-change writable
.UserID              property  y      1            emits-change writable
.Username            property  s      "root"       emits-change writable

Signed-off-by: Jayaprakash Mutyala <mutyalax.jayaprakash@intel.com>
---
 command/rakp12.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/command/rakp12.cpp b/command/rakp12.cpp
index 099c5dc..98b6891 100644
--- a/command/rakp12.cpp
+++ b/command/rakp12.cpp
@@ -227,6 +227,7 @@ std::vector<uint8_t> RAKP12(const std::vector<uint8_t>& inPayload,
     }
     session->channelNum(chNum);
     session->userID(userId);
+    session->username(userName);
     // minimum privilege of Channel / User / session::privilege::USER
     // has to be used as session current privilege level
     uint8_t minPriv = 0;
-- 
2.17.1

